### PR TITLE
usb: Remove default Vendor and Product ID

### PR DESCRIPTION
**Pull request by finikorg**
Settings configuration allows to set Vendor and Product IDs. These
configuration options should be used when building USB gadget
applications.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

### CODE REVIEW

**Comment by finikorg** [comment]
@pfl @nashif @jfischer-phytec-iot  I do not other ways to add reviewers

File: subsys/usb/Kconfig [hunk 1]
```diff
@@ -16,6 +17,24 @@ menuconfig USB_DEVICE_STACK
...
 
+menuconfig USB_DEVICE_STACK_SETTINGS
+	bool
+	prompt "USB device stack settings"
+	default n
+	help
+	Allow to set USB device stack settings including Vendor ID
+	and Product ID.
+
+if USB_DEVICE_STACK_SETTINGS
```
Comment (jfischer-phytec-iot) [inline subsys/usb/Kconfig]: It makes no sense to make it optional, PID and VID must be explicitly configured.

File: subsys/usb/Kconfig [hunk 2]
```diff
@@ -16,6 +17,24 @@ menuconfig USB_DEVICE_STACK
...
+
+config USB_DEVICE_VENDOR_ID
+	hex "USB Device Vendor ID"
+
+config USB_DEVICE_PRODUCT_ID
+	hex "USB Device Product ID"
```
Comment (jfischer-phytec-iot) [inline subsys/usb/Kconfig]: Configurable VID and PID without the ability to configure manufacturer / product strings is not really practical.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
I suggest to wait for #716 or pick 566c4cd

File: subsys/usb/Kconfig (hunk 1 above)
Comment (finikorg) [reply subsys/usb/Kconfig]: @jfischer-phytec-iot @pfl That is the idea first to introduce those parameters to make slow transition not affecting existing build sanity checks. The problem here is that we cannot assign default values here (unless you buy vendor id, in that case feel free to send a patch).

File: subsys/usb/Kconfig (hunk 2 above)
Comment (finikorg) [reply subsys/usb/Kconfig]: We can add later other values

**Comment by finikorg** [comment]
566c4cddad26a5473dc0ae189f7eeed209011c70 is not acceptable for us, those value should not be used @jfischer-phytec-iot

**Comment by nashif** [comment]
is this still relevant? We have those now as part of f0832da1.

**Comment by finikorg** [comment]
@nashif This is outdated now.

**Comment by finikorg** [comment]
@nashif @pfl I have updated this pull request

**Comment by pfl** [comment]
Looks fine. If there are other hard coded values in use, cleaning up those should make it as patch number two.

**Review by pfl — APPROVED** [review]
Looks ok to me.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
a nice try, but it's not so simple, e.g. tinytile select usb device stack by default:
boards/x86/tinytile/tinytile_defconfig:15:CONFIG_USB_UART_CONSOLE=y

**Comment by pfl** [comment]
Tinytile board file needs to contain proper VID and PID in order to be useful. Do we know already if and what values the board should use?

**Comment by finikorg** [comment]
@jfischer-phytec-iot @nashif @pfl What shall we put for tinytile? This is the board which uses USB Device stack for serial console so the parameters for Vendor and Product ID should be available.

**Review by nashif — CHANGES_REQUESTED** [review]
I am not following why are you adding prj_sanity.conf instead of changing the existing file that already exists!

**Comment by finikorg** [comment]
@nashif prj_sanity.conf configuration files are supposed to exist only for sanity check since they put some random values for Product ID and Vendor ID.

**Comment by nashif** [comment]
> @nashif prj_sanity.conf configuration files are supposed to exist only for sanity check since they put some random values for Product ID and Vendor ID.

Use the new extra_configs in testcase.yaml instead, this way we will avoid duplication of the base configuration, see #4374 for documentation of this new option.

**Comment by finikorg** [comment]
@nashif this method is not working. I have:
...
      extra_configs:
        - CONFIG_USB_DEVICE_VID=0xDEAD
        - CONFIG_USB_DEVICE_PID=0xBEEF
...

And error message is:
.config:32:warning: symbol value '0xDEADCONFIG_USB_DEVICE_PID=0xBEEF' invalid for USB_DEVICE_VID

**Comment by nashif** [comment]
found the bug, fixed in #4403

**Comment by finikorg** [comment]
@nashif fixed, the issue with tinytile remains

**Comment by nashif** [comment]
> @nashif fixed, the issue with tinytile remains

Just use whatever we have used sofar not changing the status quo here.

**Comment by finikorg** [comment]
@nashif if we could provide those 2 lines of extra_configs for tinytile sanity_check tests that would be enough. When we get Zephyr VID we assing PID to tinytile CDC ACM.

**Comment by nashif** [comment]
> @nashif if we could provide those 2 lines of extra_configs for tinytile sanity_check tests that would be enough. When we get Zephyr VID we assing PID to tinytile CDC ACM.

This has nothing to do with sanitycheck, this is a board configuration that needs to work outside of sanitychech

**Comment by jfischer-phytec-iot** [comment]
tinytile has Intel inside, can we still use Intel's VID/PID for `boards/x86/tinytile/Kconfig.defconfig`?

**Comment by finikorg** [comment]
@pfl Can we use Intel VID / PID for tinytile?

**Comment by finikorg** [comment]
@laperie @nashif

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@jfischer-phytec-iot you ok with this?

**Comment by jfischer-phytec-iot** [comment]
@nashif LGTM

**Pull request merged**